custodial_copy_notifications.tf

```diff
@@ -0,0 +1,54 @@
+locals {
+  custodial_copy_topic_name  = "${local.environment}-dr2-cc-notifications"
+  custodial_copy_topic_arn   = "arn:aws:sns:eu-west-2:${data.aws_caller_identity.current.account_id}:${local.custodial_copy_topic_name}"
+  custodial_copy_queue_name  = "${local.environment}-dr2-cc-notifications"
+  custodial_copy_lambda_name = "${local.environment}-dr2-cc-notifications"
+}
+
+module "dr2_custodial_copy_topic" {
+  source = "git::https://github.com/nationalarchives/da-terraform-modules//sns"
+  sns_policy = templatefile("./templates/sns/custodial_copy_topic_policy.json.tpl", {
+    dr_user_arn = aws_iam_user.disaster_recovery_user.arn
+    sns_topic   = local.custodial_copy_topic_arn
+  })
+  tags       = {}
+  topic_name = local.custodial_copy_topic_name
+}
+
+module "dr2_custodial_copy_queue" {
+  source     = "git::https://github.com/nationalarchives/da-terraform-modules//sqs"
+  queue_name = local.custodial_copy_queue_name
+  sqs_policy = templatefile("./templates/sqs/sns_send_message_policy.json.tpl", {
+    account_id = var.account_number,
+    queue_name = local.custodial_copy_queue_name
+    topic_arn  = local.custodial_copy_topic_arn
+  })
+  encryption_type = "sse"
+}
+
+resource "aws_sns_topic_subscription" "dr2_custodial_copy_queue_subscription" {
```

Comment:
Please can `raw_message_delivery` be enabled on this, it avoids double encapsulation when we receive the messages via SQS. I forgot to put this in the ticket.

Docs: https://docs.aws.amazon.com/sns/latest/dg/sns-large-payload-raw-message-delivery.html#raw-message-examples

Reply:
You don't need the `aws_sns_topic_subscription` resource here at all. This can be deleted and you can add a block to the topic module.
```hcl
module "dr2_custodial_copy_topic" {
  source = "git::https://github.com/nationalarchives/da-terraform-modules//sns"
  ...
  sqs_subscriptions = {
    custodial_copy_queue = module.dr2_custodial_copy_queue.sqs_arn
  }
}
```
Raw message delivery is set by default for sqs subscriptions as well.

Reply:
That's correct, that's how it was in the entity even generator which I used as template. However, I felt the SNS should not be the one initiating subscriptions hence I kept it separate. I'm not fussed about it, if you feel it's better to keep it with SNS, I can change it like that.

Reply:
Ignore the comment above, changed it to be consistent with previous